api/client/client_test.go

```diff
@@ -778,3 +778,191 @@ func TestAccessRequestDowngrade(t *testing.T) {
 	m.grpc.Stop()
 	require.NoError(t, <-remoteErr)
 }
+
+type mockRoleServer struct {
+	*mockServer
+	roles map[string]*types.RoleV5
+}
+
+func newMockRoleServer() *mockRoleServer {
+	m := &mockRoleServer{
+		&mockServer{
+			grpc:                           grpc.NewServer(),
+			UnimplementedAuthServiceServer: &proto.UnimplementedAuthServiceServer{},
+		},
+		make(map[string]*types.RoleV5),
+	}
+	proto.RegisterAuthServiceServer(m.grpc, m)
+	return m
+}
+
+func startMockRoleServer(t *testing.T) string {
+	l, err := net.Listen("tcp", "")
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, l.Close()) })
+	go newMockRoleServer().grpc.Serve(l)
+	return l.Addr().String()
+}
+
+func (m *mockRoleServer) GetRole(ctx context.Context, req *proto.GetRoleRequest) (*types.RoleV5, error) {
+	conn, ok := m.roles[req.Name]
+	if !ok {
+		return nil, trace.NotFound("not found")
+	}
+	return conn, nil
+}
+
+func (m *mockRoleServer) GetRoles(ctx context.Context, _ *empty.Empty) (*proto.GetRolesResponse, error) {
+	var connectors []*types.RoleV5
+	for _, conn := range m.roles {
+		connectors = append(connectors, conn)
+	}
+	return &proto.GetRolesResponse{
+		Roles: connectors,
+	}, nil
+}
+
+func (m *mockRoleServer) UpsertRole(ctx context.Context, role *types.RoleV5) (*empty.Empty, error) {
+	m.roles[role.Metadata.Name] = role
+	return &empty.Empty{}, nil
+}
+
+func (m *mockRoleServer) GetCurrentUserRoles(_ *empty.Empty, stream proto.AuthService_GetCurrentUserRolesServer) error {
+	for _, role := range m.roles {
+		if err := stream.Send(role); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+// Test that client will perform properly with an old server
+// DELETE IN 13.0.0
+func TestSetRoleRequireSessionMFABackwardsCompatibility(t *testing.T) {
+	ctx := context.Background()
+	addr := startMockRoleServer(t)
+
+	// Create client
+	clt, err := New(ctx, Config{
+		Addrs: []string{addr},
+		Credentials: []Credentials{
+			&mockInsecureTLSCredentials{}, // TODO(Joerger) replace insecure credentials
+		},
+		DialOpts: []grpc.DialOption{
+			grpc.WithTransportCredentials(insecure.NewCredentials()), // TODO(Joerger) remove insecure dial option
+		},
+	})
+	require.NoError(t, err)
+
+	role := &types.RoleV5{
+		Metadata: types.Metadata{
+			Name: "one",
+		},
+	}
+
+	// Upsert should set "RequireSessionMFA" on the provided role if "RequireMFAType" is set
+	role.Spec.Options.RequireMFAType = types.RequireSessionMFA
+	role.Spec.Options.RequireSessionMFA = false
+	err = clt.UpsertRole(ctx, role)
+	require.NoError(t, err)
+	require.Equal(t, role.GetOptions().RequireMFAType.IsSessionMFARequired(), role.GetOptions().RequireSessionMFA)
```

Comment:
```suggestion
	// Upsert should set "RequireSessionMFA" on the provided role if "RequireMFAType" is set
	role.Spec.Options.RequireMFAType = types.RequireSessionMFA
	role.Spec.Options.RequireSessionMFA = false
	err = clt.UpsertRole(ctx, role)
	require.NoError(t, err)
	require.True(t, role.GetOptions().RequireSessionMFA)
```

I think this test should just assert `role.GetOptions().RequireSessionMFA` equals `true` directly. Calling `IsSessionMFARequired()` introduces extra logic to the test assertion.